app/forms/supplier_user_form.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+class SupplierUserForm
+  include ActiveModel::Model
+  include ActiveModel::Serialization
+
+  attr_accessor :full_name, :email, :supplier
+  validates :supplier, presence: { message: "Select one" }, on: :supplier
+  validates :full_name, presence: { message: "Enter a name" }, on: :details
+  validates :email, presence: { message: "Enter email" }, on: :details
+  validate :email_not_taken, on: :details
+
+  def attributes
+    { "full_name" => nil, "email" => nil, "supplier" => nil }
+  end
+
+  def chosen_supplier
+    LeadProvider.find_by(id: supplier) || DeliveryPartner.find_by(id: supplier)
+  end
+
+  def save!
+    profile = if chosen_supplier.instance_of?(LeadProvider)
+                LeadProviderProfile.new(lead_provider: chosen_supplier)
+              else
+                DeliveryPartnerProfile.new(delivery_partner: chosen_supplier)
+              end
+
+    ActiveRecord::Base.transaction do
+      @user = User.create!(
+        full_name: full_name,
+        email: email,
+        confirmed_at: Time.zone.now.utc, # Skip confirmation email
+      )
+      profile.user = @user
+      profile.save!
+    end
+    @user
```

Comment:
Does this need to be a `@` variable?

Reply:
Yes, since I create it inside the transaction block and want to return it. Out of scope otherwise

Reply:
Will something like 
```
    user = ActiveRecord::Base.transaction do
      user = User.create!(
        full_name: full_name,
        email: email,
        confirmed_at: Time.zone.now.utc, # Skip confirmation email
      )
      profile.user = user
      profile.save!
      user
    end
    user
```
work?